test/Libraries/Microsoft.Extensions.TimeProvider.Testing.Tests/FakeTimeProviderTests.cs

```diff
@@ -381,4 +381,57 @@ public void ShouldResetGateUnderLock_PreventingContextSwitching_AffectionOnTimer
         // Assert
         Assert.Single(calls);
     }
+
+    [Fact]
+    public void SimulateRetryPolicy()
+    {
+        // Arrange
+        var retries = 42;
+        var tries = 0;
+        var taskDelay = 0.5;
+        var delay = 1;
+        var provider = new FakeTimeProvider();
+
+        async Task<int> simulatedPollyRetry()
+        {
+            while (true)
+            {
+                try
+                {
+                    // simulate task that takes some time to complete
+                    await provider.Delay(TimeSpan.FromSeconds(taskDelay));
+                    tries++;
+
+                    if (tries <= retries)
+                    {
+                        // the task failed, trigger retry
+                        throw new InvalidOperationException();
+                    }
+
+                    return tries;
+                }
+                catch (InvalidOperationException)
+                {
+                    // ConfigureAwait(true) is required to ensure that tasks continue on the captured context
+                    await provider.Delay(TimeSpan.FromSeconds(delay)).ConfigureAwait(true);
+                }
+            }
+        }
+
+        // Act
+        var result = simulatedPollyRetry();
```

Comment:
Please don't use `var` unless the type is obvious.
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md #<!-- -->10